src/main/java/org/usfirst/frc/team4099/robot/subsystems/Arm.kt

```diff
@@ -0,0 +1,175 @@
+    }
+
+
+
+
+    init {
+        masterSRX.set(ControlMode.MotionMagic, 0.0)
+        slaveSRX1.set(ControlMode.Follower, Constants.Arm.MASTER_SRX_ID.toDouble())
+        slaveSRX2.set(ControlMode.Follower, Constants.Arm.MASTER_SRX_ID.toDouble())
+        slaveSRX3.set(ControlMode.Follower, Constants.Arm.MASTER_SRX_ID.toDouble())
+        masterSRX.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, 10)
+        masterSRX.setSensorPhase(false)
+        masterSRX.configNominalOutputForward(+0.0, 0)
+        masterSRX.configPeakOutputForward(+1.0, 0)
+        masterSRX.selectProfileSlot(0, 0)
+        masterSRX.config_kF(0, 0.0, 0)
+        masterSRX.config_kP(0, 0.0, 0)
+        masterSRX.config_kI(0, 0.0, 0)
+        masterSRX.config_kD(0, 0.0, 0)
+        armBaseAngle = ArmConversion.pulsesToRadians(masterSRX.sensorCollection.pulseWidthPosition)
+
+    }
+
+
+    /**
+     * Changes power to the arm
+     *
+     * @param power A double for the power going to the arm
+     */
+    private fun setArmPower(power: Double) {
+        armState = ArmState.VELOCITY_CONTROL
+        masterSRX.set(ControlMode.MotionMagic, Math.abs(power))
+        armPower = power
+    }
+
+    /**
+     * Outputs arm subsystem
+     *
+     */
+    override fun outputToSmartDashboard() {
+        SmartDashboard.putNumber("armPower", armPower)
+    }
+
+    @Synchronized override fun stop() {
+        movementState = MovementState.STATIONARY
+        setArmPower(0.0)
+    }
+
+    /**
+     * Handles power to arm during each loop cycle
+     *
+     * @constructor Creates loop that controls power to arm during each loop cycle
+     */
+    val loop: Loop = object : Loop {
+
+        override fun onStart() {
+            movementState = MovementState.STATIONARY
+            setArmPower(0.0)
+        }
+
+        /**
+         * Handles events during each loop cycle, depends on arm control mode
+         */
+        override fun onLoop() {
+            synchronized(this@Arm) {
+                when (armState) {
+                    Arm.ArmState.LOW -> {
+                        masterSRX.set(ControlMode.MotionMagic, ArmConversion.radiansToPulses(ArmState.LOW.targetPos).toDouble())
+                        armAngle = ArmState.LOW.targetPos
+                    }
+                    Arm.ArmState.EXCHANGE -> {
+                        masterSRX.set(ControlMode.MotionMagic, ArmConversion.radiansToPulses(ArmState.EXCHANGE.targetPos).toDouble())
+                        armAngle = ArmState.EXCHANGE.targetPos
+                    }
+                    Arm.ArmState.HIGH -> {
+                        masterSRX.set(ControlMode.MotionMagic, ArmConversion.radiansToPulses(ArmState.HIGH.targetPos).toDouble())
+                        armAngle = ArmState.HIGH.targetPos
+                    }
+                    ArmState.VELOCITY_CONTROL -> {
+                        masterSRX.set(ControlMode.Velocity, 0.0)
+                        when (movementState) {
```

Comment:
movement states should be set regardless of whether or not it's in velocity control.

Also, in velocity control, more than just the state will be set -- have it literally just `pass` in this case, and let the setPower function be public (in velocity control mode, use that function to control movement and direction)